python/ray/rllib/agents/agent.py

```diff
@@ -545,7 +559,8 @@ def _validate_config(config):
                 "`input_evaluation` should not be set when input=sampler")
 
     def _make_evaluator(self, cls, env_creator, policy_graph, worker_index,
-                        config):
+                        config, remote_worker_envs=False,
+                        entangled_worker_envs=False):
```

Comment:
These args can be passed as part of `config` right?

Reply:
Removed entangled_worker_envs.

remote_worker_envs is hardcoded to False for local_evaluator, so it doesn't create ray workers if it doesn't need to. Also, you can see that dummy env in each evaluator is created with remote false with the same reason.

A lot of problems boils down to the issue that env needs to be created for dummy values. I encourage opening new environment register interface as mentioned in pull request description, and obsoleting current one (printing obsolete when used and removing it after few releases)...

Reply:
I see. So one reason it makes sense to put under config is that you might want to run with `num_workers: 0` with remote envs.

I think you can instead have the remote env wrapper initialize the remote workers only on reset(), similar to the rest of the env adapters. I agree the registration interface could use re-thinking though for the future.

Reply:
I don't get how / why would you run with num_workers 0? Local evaluator is used for optimizer and remote evaluators are used for sampling, why would you want to mix those?

Reply:
Local environment currently creates num num_envs_per_worker environments too much (it should not create neither one). It would be too costly to request this much ray workers and could even lead to hang if they couldn't be scheduled.